proposals/buildfile.md

```diff
@@ -0,0 +1,128 @@
+# Jib CLI Buildfile Specification
+
+Specification for a YAML buildfile describing building a container image. This buildfile can be
+used by the jib-cli to generate a container. It is translated directly into a buildplan and
+passed to the builder.
+
+```
+# "FROM" with option to use os/architecture for manifest lists
+baseImage: "gcr.io/distroless/java:8"
+baseImageOs: linux
```

Comment:
There are other properties to select an image from a manifest list / [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md).  Maybe this can be made more explicit as an map.  Something like:
```
from:
  image: "gcr.io/distroless/java:8"
  platform:
    os: linux
    arch: arm
    variant:  v6
```

It would be very useful to try expressing your format as a JSON schema so that it can be used to provide auto-complete support by IDEs.  @dgageot wrote some scripts for [Skaffold to turn Go-based structures into JSON schema](https://github.com/GoogleContainerTools/skaffold/blob/master/hack/schemas/main.go).

Reply:
reworked this section.